gui/services/forms.py

```diff
@@ -439,30 +439,35 @@ def save(self):
         return obj
 
 
-class RsyncdForm(ModelForm):
+class RsyncdForm(MiddlewareModelForm, ModelForm):
+
+    middleware_attr_prefix = "rsyncd_"
+    middleware_attr_schema = "rsyncd"
+    middleware_plugin = "rsyncd"
 
     class Meta:
         fields = '__all__'
         model = models.Rsyncd
 
-    def save(self):
-        super(RsyncdForm, self).save()
-        started = notifier().reload("rsync")
-        if (
-            started is False and
-            models.services.objects.get(srv_service='rsync').srv_enable
-        ):
-            raise ServiceFailed(
-                "rsync", _("The Rsync service failed to reload.")
-            )
-
 
 class RsyncModForm(ModelForm):
 
     class Meta:
         fields = '__all__'
         model = models.RsyncMod
 
+    def save(self):
+        with client as c:
+            cdata = self.cleaned_data
+
+            if self.instance.id:
+                c.call('rsyncdmod.do_update', self.instance.id, cdata)
+            else:
+                c.call('rsyncdmod.do_create', cdata)
+            c.call('service.reload', 'rsync')
```

Comment:
Don't do `service.reload` here. Do it in the middleware. Otherwise this logic will be lost in new GUI.